tensorboard/webapp/widgets/data_table/filter_dialog_test.ts

```diff
@@ -0,0 +1,143 @@
+    fixture.componentInstance.filter = input.filter;
+    rootLoader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+    fixture.detectChanges();
+    return fixture;
+  }
+
+  it('dispatches event when an interval filter value is changed', async () => {
+    const fixture = createComponent({
+      filter: {
+        type: DomainType.INTERVAL,
+        includeUndefined: false,
+        minValue: 6,
+        maxValue: 20,
+        filterLowerValue: 7,
+        filterUpperValue: 18,
+      },
+    });
+
+    const intervalFilterChangedSpy = spyOn(
+      fixture.componentInstance.intervalFilterChanged,
+      'emit'
+    );
+
+    const rangeInput = fixture.debugElement.query(
+      By.directive(RangeInputComponent)
+    );
+    rangeInput.componentInstance.rangeValuesChanged.emit({
+      lowerValue: 7,
+      upperValue: 17,
+    });
+    expect(intervalFilterChangedSpy).toHaveBeenCalled();
+
+    const includeUndefinedSpy = spyOn(
+      fixture.componentInstance.includeUndefinedToggled,
+      'emit'
+    );
+
+    const includeUndefinedCheckbox = await rootLoader.getHarness(
+      MatCheckboxHarness.with({label: 'Include Undefined'})
+    );
+    await includeUndefinedCheckbox.check();
+    expect(includeUndefinedSpy).toHaveBeenCalled();
+  });
+
+  it('dispatches event when an discrete filter value is changed', async () => {
+    const fixture = createComponent({
+      filter: {
+        type: DomainType.DISCRETE,
+        includeUndefined: false,
+        possibleValues: [2, 4, 6, 8],
+        filterValues: [2, 4, 6],
+      },
+    });
+    const discreteFilterChangedSpy = spyOn(
+      fixture.componentInstance.discreteFilterChanged,
+      'emit'
+    );
+    const checkbox = await rootLoader.getHarness(
+      MatCheckboxHarness.with({label: '2'})
+    );
+    await checkbox.uncheck();
+    fixture.debugElement
+      .query(By.css('mat-checkbox label'))
+      .nativeElement.click();
+    expect(discreteFilterChangedSpy).toHaveBeenCalled();
+
+    const includeUndefinedSpy = spyOn(
+      fixture.componentInstance.includeUndefinedToggled,
+      'emit'
+    );
+
+    const includeUndefinedCheckbox = await rootLoader.getHarness(
+      MatCheckboxHarness.with({label: 'Include Undefined'})
+    );
+    await includeUndefinedCheckbox.check();
+    expect(includeUndefinedSpy).toHaveBeenCalled();
+  });
+
+  it('filters discrete values', async () => {
+    const fixture = createComponent({
+      filter: {
+        type: DomainType.DISCRETE,
+        includeUndefined: false,
+        possibleValues: ['foo', 'bar', 'baz', 'qaz'],
+        filterValues: ['foo', 'bar', 'baz', 'qaz'],
+      },
+    });
+    expect(
+      (await rootLoader.getAllHarnesses(MatCheckboxHarness)).length
+    ).toEqual(5); // 4 options + the include undefined checkbox
+
+    fixture.componentInstance.discreteValueFilter = 'ba';
+    fixture.detectChanges();
+    expect(
+      (await rootLoader.getAllHarnesses(MatCheckboxHarness)).length
```

Comment:
I would be more confident in the test if you actually checked that it's the 'bar' and 'baz' boxes shown.

Reply:
This would still make me more confident. (You resolved this but I didn't see corresponding changes).

Reply:
I'm not sure how this has happened but this comment is on an outdated version of the file.

Clicking your comment allows me to see the code you are commenting on:
![image](https://github.com/tensorflow/tensorboard/assets/78179109/862ede55-3618-4ad0-bc12-19d68ac888df)

However, clicking the "Files Changed" tab I see a much different file (where I did address your comment).
![image](https://github.com/tensorflow/tensorboard/assets/78179109/acb176cd-8ddb-4c83-b993-93cd75906fc0)

![image](https://github.com/tensorflow/tensorboard/assets/78179109/3785306c-3090-4f73-a286-36d72ebd54c8)